src/modules/history/HistoryFilterMenu.tsx

```diff
@@ -0,0 +1,258 @@
+import { XCircleIcon } from '@heroicons/react/solid';
+import { Trans } from '@lingui/macro';
+import { Check as CheckIcon, Sort as SortIcon } from '@mui/icons-material';
+import {
+  Box,
+  Button,
+  Divider,
+  Menu,
+  MenuItem,
+  SvgIcon,
+  Typography,
+  useMediaQuery,
+  useTheme,
+} from '@mui/material';
+import React, { useEffect, useState } from 'react';
+import { DarkTooltip } from 'src/components/infoTooltips/DarkTooltip';
+
+import { FilterOptions } from './types';
+
+interface HistoryFilterMenuProps {
+  onFilterChange: (filter: FilterOptions[]) => void;
+  currentFilter: FilterOptions[];
+}
+
+interface FilterLabelProps {
+  filter: FilterOptions;
+}
+
+const FilterLabel: React.FC<FilterLabelProps> = ({ filter }) => {
+  switch (filter) {
+    case FilterOptions.SUPPLY:
+      return <Trans>Supply</Trans>;
+    case FilterOptions.BORROW:
+      return <Trans>Borrow</Trans>;
+    case FilterOptions.WITHDRAW:
+      return <Trans>Withdraw</Trans>;
+    case FilterOptions.REPAY:
+      return <Trans>Repay</Trans>;
+    case FilterOptions.RATECHANGE:
+      return <Trans>Rate change</Trans>;
+    case FilterOptions.COLLATERALCHANGE:
+      return <Trans>Collateral change</Trans>;
+    case FilterOptions.LIQUIDATION:
+      return <Trans>Liqudation</Trans>;
```

Comment:
Typo here, i don't add a code suggestion because it needs building for the translations